src/util/util_errors.h

```diff
@@ -170,7 +170,7 @@ enum sssd_errors {
 #define SSSD_ERR_BASE(err) ((err) & ~ERR_MASK)
 #define SSSD_ERR_IDX(err) ((err) & ERR_MASK)
 #define IS_SSSD_ERROR(err) \
-    ((SSSD_ERR_BASE(err) == ERR_BASE) && ((err) <= ERR_LAST))
+    (((err) > 0) && (SSSD_ERR_BASE(err) == ERR_BASE) && ((err) <= ERR_LAST))
```

Comment:
I'm not sure if I understand this fix. Covscan complains of using a negative parameter for `sss_ldap_err2string()` when it cannot be negative. But you are fixing one of its callees `IS_SSSD_ERROR()`, which will change the flow path from calling `sss_strerror()` to call `ldap_err2string()` instead.

Reply:
Covscan complains bitwise operation (`SSSD_ERR_BASE`) is applied to negative number and have a reason to do so.

Negative number can't be sssd_error.

Flow path is correctly ensured with this patch.

Reply:
Now it makes sense. Thanks for the clarification.